internal/repositories/service.go

```diff
@@ -228,18 +228,28 @@ func (r *repositoryService) CreateRepository(
 func (r *repositoryService) ListRepositories(
 	ctx context.Context,
 	projectID uuid.UUID,
-	providerName string,
-) ([]db.Repository, error) {
-	return r.store.ListRepositoriesByProjectID(
-		ctx,
-		db.ListRepositoriesByProjectIDParams{
-			ProjectID: projectID,
-			Provider: sql.NullString{
-				String: providerName,
-				Valid:  providerName != "",
-			},
-		},
-	)
+	providerID uuid.UUID,
+) ([]*models.EntityWithProperties, error) {
+	repoEnts, err := r.store.GetEntitiesByType(ctx, db.GetEntitiesByTypeParams{
+		EntityType: db.EntitiesRepository,
+		ProviderID: providerID,
+		Projects:   []uuid.UUID{projectID},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error fetching repositories: %w", err)
+	}
+
+	ents := make([]*models.EntityWithProperties, 0, len(repoEnts))
+	for _, ent := range repoEnts {
+		ewp, err := r.propSvc.EntityWithProperties(ctx, ent.ID, nil)
```

Comment:
should this be a transaction so that handlers are not interrupted in case the repo goes away?